README.md

```diff
@@ -139,76 +252,93 @@ let attributesList = [stringAttribute, intAttribute, boolAttribute, doubleAttrib
+RInAppMessaging.closeMessage(clearQueuedCampaigns: true)
+```
+**Note:** Calling this API will not increment the campaign's impression (i.e. not counted as displayed).
 
-⚠️ The method is designed to be called ONCE once per app session - i.e. only one instance of `UserInfoProvider` can be created. IAM SDK will read object's properties on demand. There's no need to call this method again after login/logout for example.
+## <a name="final-code"></a>Final Code (Sample)
 
-To ensure correct user targeting, please keep user information in the preference object up to date.
-After logout process is complete the preference object should return `nil` or `""` in all `UserInfoProvider` methods.  
-Preferences are not persisted so this function needs to be called on every launch.
+**For reference, your SDK integration code should look something like this:**
 
-#### Generic example using UserID
+<details>
+<summary style="cursor: pointer;";>(click to expand)</summary>
 
+AppDelegate.swift
 ```swift
 import RInAppMessaging
 
-class UserPreference: UserInfoProvider {
-    func getUserID() -> String? { 
-        "member-id" 
-    }
-
-    func getIDTrackingIdentifier() -> String? { nil }
-    func getAccessToken() -> String? { nil }
+func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplicationLaunchOptionsKey: Any]?) -> Bool {
+    RInAppMessaging.configure(subscriptionID: "subscriptionID", 
+                                configurationURL: "configURL",
+                                enableTooltipFeature: true)
+    RInAppMessaging.registerPreference(self.preferenceData)
+    return true
 }
-
-let preference = UserPreference()
-RInAppMessaging.registerPreference(preference)
 ```
 
 
-### **closeMessage()**
-
-In certain cases there might be a need to manually close a campaign's message without user interaction.
-An example is when a different user logs in and the currently displayed campaign does not target the new user.
-(Or when a campaign's message appears after login process has started).
-In that case, to avoid user's confusion, host app can force-close the campaign by calling `closeMessage()` API method.
-The `clearQueuedCampaigns` optional parameter, when set to `true` (`false` by default), will additionally remove all campaigns that were queued to be displayed.
-
+IAMPreferenceData.swift
 ```swift
-RInAppMessaging.closeMessage(clearQueuedCampaigns: true)
-```
-**Note:** Calling this API will not increment the campaign's impression (i.e. not counted as displayed).
+import RInAppMessaging
 
+class IAMPreferenceData: UserInfoProvider {
+    var userID: String?
+    var easyID: String?
+    var accessToken: String?
 
-## **Custom Events**
+}
 
-As shown in the example above for event logging, IAM also supports custom event. Custom events are events with an unique name and a list of attributes associated with them.
+extension IAMPreferenceData {
+    func getAccessToken() -> String? {
+        "accessToken"
```

Comment:
I guess, we need to return the instance not String. else there is no use of declaring the properties in the `IAMPreferenceData`